lib/assessment.js

```diff
@@ -8,15 +8,21 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
-
+var min = function min(previousValue, currentValue, index, array) {
+  var minimum = previousValue;
+  for (var i=0; i<array.length; i++){
+    if (currentValue < minimum){
+      minimum = currentValue;
+    };
+  }return minimum;
```

Comment:
What keeps your loop from working is that the loop index variable is not used inside the body. If the body of a loop is not dependent on the loop index variable, then it does the exact same thing each time without variation. In this case, it tests the exact same `currentValue` against `minimum` and, if the test succeeds, it assigns the exact same `currentValue` to `minimum`.

Now, that's what is making the loop inoperative. The real problem here is that you shouldn't be looping in the first place. This is meant to be a callback for the `Array.forEach` method, which does all the looping for you. The callback's role is simply to test the parameters for certain conditions (e.g., `currentValue < previousValue`) and return `true` for one case and `false` otherwise.

Half credit.